include/no_os_clk.h

```diff
@@ -47,6 +47,17 @@
 /******************************************************************************/
 /************************* Structure Declarations *****************************/
 /******************************************************************************/
+struct no_os_clk_init_param {
+	/** Device name */
+	const char	*name;
+	/** Channel number */
+	uint8_t		hw_ch_num;
+	/** CLK function pointers */
+	const struct no_os_clk_platform_ops *platform_ops;
+	/**  CLK hardware device descriptor */
+	void		*dev_desc;
+};
+
 struct no_os_clk_hw {
```

Comment:
I think that following this change, struct no_os_clk_hw might not be needed.

Reply:
This was kept so that current implementation does not suffer.